### ISSUE TITLE
windows user.present broken

### ISSUE THREAD

**Issue by jfindlay** [saltstack/salt#26336]
### configs
```saltstack
# cat tests/windows/user_add.sls
add_dude:
  user.present:
    - name: saltdude
    - fullname: "Salt Dude"
```

### commands

First run:
```console
# salt jmoney-windows-12 state.sls tests.windows.user_add
jmoney-windows-12:
----------
          ID: add_dude
    Function: user.present
        Name: saltdude
      Result: True
     Comment: New user saltdude created
     Started: 19:18:32.769000
    Duration: 359.0 ms
     Changes:
              ----------
              active:
                  True
              comment:
              fullname:
                  Salt Dude
              gid:
              groups:
                  - Users
              home:
                  ----------
                  comment:
                      [Error 2] The system cannot find the file specified
                  hive:
                      HKEY_LOCAL_MACHINE
                  key:
                      SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-21-1853168970-4164690662-1339155858-1001
                  success:
                      False
                  vdata:
                      None
                  vname:
                      ProfileImagePath
              logonscript:
              name:
                  saltdude
              passwd:
                  None
              profile:
                  ----------
                  comment:
                      [Error 2] The system cannot find the file specified
                  hive:
                      HKEY_LOCAL_MACHINE
                  key:
                      SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-21-1853168970-4164690662-1339155858-1001
                  success:
                      False
                  vdata:
                      None
                  vname:
                      ProfileImagePath
              uid:
                  S-1-5-21-1853168970-4164690662-1339155858-1001

Summary
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
```

Second run:
```console
# salt jmoney-windows-12 state.sls tests.windows.user_add
jmoney-windows-12:
----------
          ID: add_dude
    Function: user.present
        Name: saltdude
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "c:\salt\bin\lib\site-packages\salt\state.py", line 1561, in call
                  **cdata['kwargs'])
                File "c:\salt\bin\lib\site-packages\salt\states\user.py", line 376, in present
                  expire)
                File "c:\salt\bin\lib\site-packages\salt\states\user.py", line 114, in _changes
                  if newhome is not None and not os.path.isdir(newhome):
              TypeError: coercing to Unicode: need string or buffer, dict found
     Started: 19:18:45.597000
    Duration: 16.0 ms
     Changes:

Summary
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
ERROR: Minions returned with non-zero exit code
```

### versions
```yaml
# salt jmoney-windows-12 test.versions_report
jmoney-windows-12:
               Salt: 2015.5.3-235-g1e3f33a
             Python: 2.7.8 (default, Jun 30 2014, 16:08:48) [MSC v.1500 64 bit (AMD64)]
             Jinja2: 2.7.3
           M2Crypto: 0.21.1
     msgpack-python: 0.4.5
       msgpack-pure: Not Installed
           pycrypto: 2.6
            libnacl: Not Installed
             PyYAML: 3.11
              ioflo: Not Installed
              PyZMQ: 14.6.0
               RAET: Not Installed
                ZMQ: 4.0.5
               Mako: Not Installed
            Tornado: Not Installed
            timelib: Not Installed
```

**Comment by twangboy** [comment]
@jfindlay I think this has been fixed. I just ran on the 2015.5.5 tag and I can run user.present twice with no issue.

**Comment by jfindlay** [comment]
@twangboy, do you get the `[Error 2] The system cannot find the file specified` comment?

**Comment by twangboy** [comment]
@jfindlay, no. It just works for me. First time showing changes. Second time showing that it's already in the correct state.

**Comment by twangboy** [comment]
```
rockkit:
  user.present:                                                                                                       
   - createhome: True                                                                                                   
   - fullname: The Rocketeerr                                                                                           
   - groups:                                                                                                            
     - Administrators                                                                                                   
   - password: 'Sn00pRockx'
```

**Comment by twangboy** [comment]
There was a problem with the way it was accessing the registry to determine the profile directory. That has been fixed in PR: #28003. Will be applied to 2015.8.2.
